scripts/generate-install-yaml.sh

```diff
@@ -20,10 +20,33 @@ set -o pipefail
 
 INSTALL_YAML="${INSTALL_YAML:-hack/install-latest.yaml}"
 IMAGE_NAME="${IMAGE_NAME:-quay.io/kubernetes-multicluster/federation-v2:latest}"
+FEDERATION_NAMESPACE="${NS:-""}"
 
 kubebuilder create config --controller-image "${IMAGE_NAME}" \
             --name federation --output "${INSTALL_YAML}"
 
 # Increase memory request and limit to avoid OOM issues.
 sed -i 's/memory: 20Mi/memory: 64Mi/' "${INSTALL_YAML}"
 sed -i 's/memory: 30Mi/memory: 128Mi/' "${INSTALL_YAML}"
+
+# Remove namespace resource from the top of the file
+sed -i -e '/---/,$!d' "${INSTALL_YAML}"
+
+# Remove namespace fields
+sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
+
+if [ "${FEDERATION_NAMESPACE}" ]; then
+  # Update name for ClusterRole
+  sed -i -e 's/name: federation-role$/name: federation-role-'${FEDERATION_NAMESPACE}'/' "${INSTALL_YAML}"
+
+  # Update name for ClusterRoleBinding
+  sed -i -e 's/name: federation-rolebinding$/name: federation-rolebinding-'${FEDERATION_NAMESPACE}'/' "${INSTALL_YAML}"
+
+  # Add namespace to ClusterRoleBinding
+  sed -i -e '/- kind: ServiceAccount/a\
+  namespace: '${FEDERATION_NAMESPACE}'' "${INSTALL_YAML}"
```

Comment:
Why is this desirable for a namespaced deployment?  The removal of lines 29-30 in `scripts/generate-namespaced-install-yaml.sh` will preclude this field from being removed.

Reply:
This is because clusterrolebindings request a namespace, please refer to https://github.com/kubernetes-sigs/federation-v2/pull/146